app/models/vaccine.rb

```diff
@@ -21,15 +22,15 @@ class Vaccine < ApplicationRecord
   # --- FIELD VALIDATION --- #
   validates :group_name, inclusion: {
     # This method syntax is necessary for the getter method to be in scope
-    in: ->(_vaccine) { group_name_options },
+    in: ->(_vaccine) { [*group_name_options, nil] },
     message: lambda { |_vaccine, _data|
       "is not an acceptable value, acceptable values are: '#{group_name_options.join("', '")}'"
     }
   }, presence: { message: 'is required' }
 
   # Product name valid options depend on the current group name
   validates :product_name, inclusion: {
-    in: ->(vaccine) { product_name_options(vaccine[:group_name]) },
+    in: ->(vaccine) { [*product_name_options(vaccine[:group_name]), nil] },
```

Comment:
Same comment as above but for product name `bundle exec rails test -n test_validates_product_name`

Reply:
https://github.com/SaraAlert/SaraAlert/pull/991/files#diff-55f69eadeff0a340b9e0c84bf77a5e21f1745e101319cc8603ab2c8b9f4f107eR47